internal/resource/resource.go

```diff
@@ -155,7 +157,12 @@ func NewResource(ctx context.Context, renv *readiness.Env, slice *apiv1.Resource
 	res.DisableUpdates = anno[disableUpdatesKey] == "true"
 	delete(anno, disableUpdatesKey)
 
-	for key, value := range parsed.GetAnnotations() {
+	const readinessGroupKey = "eno.azure.io/readiness-group"
+	rg, _ := strconv.ParseUint(anno[readinessGroupKey], 10, 0)
```

Comment:
Why not bitSize 8 and handle the error? Also, why make the readiness groups so limited? Performance?

Reply:
My initial implementation just bruteforced the readiness groups, so keeping the space small was significant. But I have since swapped that out for a red/black tree so we might as well use a "full sized" uint.

Also added code to log the errors, what do you think about continuing vs failing when things fail to parse? My concern is that we don't have any notion of terminal synth errors that cannot be fixed with retries, so we'd end up stuck in a synthesis loop.

Reply:
Without a notion or terminal errors, I think continuing and just defaulting to 0 is the best we can do.